Readme.md

```diff
@@ -1,31 +1,79 @@
+This example demonstrates how display a custom "Task details" dialog instead of the default dialog. 
+
+## Implementation Details
+
+1. Add a popup edit form in your application.
    
-        function onTaskEditDialogShowing(e) { 
-			e.cancel = true;
-        	 ...  } 
-		 
-2. Create an edit form using Popup. Display it and bind it to an edited task data.
- 
-        function showTaskDetails(data) { 
-			popup.option("visible", true);
-        	if (form)
-           	 form.option('formData', data);  } 
-		 
-3.  Use the [updateTask](https://js.devexpress.com/Documentation/ApiReference/UI_Widgets/dxGantt/Methods/#updateTaskkey_data) and [assignResourceToTask](https://js.devexpress.com/Documentation/ApiReference/UI_Widgets/dxGantt/Methods/#assignResourceToTaskresourceKey_taskKey)/[unassignResourceFromTask](https://js.devexpress.com/Documentation/ApiReference/UI_Widgets/dxGantt/Methods/#unassignResourceFromTaskresourceKey_taskKey) method to update Gantt data. 
-
-Files to look at: 
-
- [Index.cshtml](https://github.com/DevExpress-Examples/devextreme-gantt--how-to-create-a-custom-task-details-dialog/blob/20.2.3%2B/CS/DevExtremeMvcApp1/Views/Home/Index.cshtml "Index.cshtml")
-
-[SampleDataController.cs](https://github.com/DevExpress-Examples/devextreme-gantt--how-to-create-a-custom-task-details-dialog/blob/20.2.3%2B/CS/DevExtremeMvcApp1/Controllers/SampleDataController.cs "SampleDataController.cs")
-
- [GanttDataProvider.cs](https://github.com/DevExpress-Examples/devextreme-gantt--how-to-create-a-custom-task-details-dialog/blob/20.2.3%2B/CS/DevExtremeMvcApp1/Models/GanttDataProvider.cs "GanttDataProvider.cs")
+	```csharp
+	@(Html.DevExtreme().Popup()
+	    .ID("taskDetailsPopup").MaxWidth(800).MaxHeight(500).Title("Task Details")
+	    .ContentTemplate(new TemplateName("customPopupContentTemplate"))
+	    .ToolbarItems(items => {
+	        items.Add()
+	            .Widget(editor => editor.Button()
+	                .Text("Confirm")
+	                .Type(ButtonType.Success)
+	                .OnClick("onConfirmClick")
+	            )
+	            .Location(ToolbarItemLocation.After)
+	            .Toolbar(Toolbar.Bottom);
+	        items.Add()
+	            .Widget(editor => editor.Button()
+	                .Text("Cancel")
+	                .Type(ButtonType.Success)
+	                .OnClick("onCancelClick")
+	            )
+	            .Location(ToolbarItemLocation.After)
+	            .Toolbar(Toolbar.Bottom);
+	    })
+	    .OnInitialized("onPopupInitialized").OnShown("onShown")
+	)
+ 	```
+	```jscript
+    function onPopupInitialized(e) {
+        popup = e.component;
+    }
+ 	```
+
+2. Handle the [taskEditDialogShowing](https://js.devexpress.com/jQuery/Documentation/ApiReference/UI_Components/dxGantt/Events/#taskEditDialogShowing) event to prevent the default dialog and display your custom dialog instead. Bind the form in the popup control to the processed task data.
```

Comment:
```suggestion
2. Handle the [taskEditDialogShowing](https://js.devexpress.com/jQuery/Documentation/ApiReference/UI_Components/dxGantt/Events/#taskEditDialogShowing) event to prevent the default dialog and display your custom dialog instead. Bind the form in the popup control to processed task data.
```